### PR TITLE
meow: init at 2.1.3

### PR DESCRIPTION
**Pull request by PixelSergey**
## Description of changes

`meow` -  Print ASCII cats to your terminal

This is a simple package that outputs a random ASCII drawing of a cat when executed.

```
Usage: meow [OPTIONS]

Options:
  -c, --count <COUNT>  How many cats to print [default: 1]
  -l, --literally      Are you literally this cat?
  -h, --help           Print help
  -V, --version        Print version
```

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/me/meow/package.nix
Comment (kilianar) [inline pkgs/by-name/me/meow/package.nix:34]: You should add yourself to the list of maintainers here: [maintainer-list.nix](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix). After that, include your name in the `meta.maintainers` list like this:

```nix
maintainers = with lib.maintainers; [ pixelsergey ];
```

File: pkgs/by-name/me/meow/package.nix
Comment (kilianar) [inline pkgs/by-name/me/meow/package.nix:31]:
```suggestion
    description = "Print ASCII cats to your terminal";
```

File: pkgs/by-name/me/meow/package.nix
Comment (PixelSergey) [reply pkgs/by-name/me/meow/package.nix]: Fixed!

File: pkgs/by-name/me/meow/package.nix
Comment (PixelSergey) [reply pkgs/by-name/me/meow/package.nix]: Thanks, fixed!

File: pkgs/by-name/me/meow/package.nix
Comment (kilianar) [inline pkgs/by-name/me/meow/package.nix:8]: The additional arguments (`pkgs`, `config`, `...`) are not used and should be removed.

```suggestion
{
  lib,
  fetchFromGitHub,
  rustPlatform,
}:
```

File: nixos/doc/manual/release-notes/rl-2411.section.md
Comment (Aleksanaa) [inline nixos/doc/manual/release-notes/rl-2411.section.md:129]: This is for NixOS modules, not every packages, otherwise we'll have tons of them

File: pkgs/by-name/me/meow/package.nix
Comment (kilianar) [inline pkgs/by-name/me/meow/package.nix:18]:
```suggestion
  version = "2.1.3";

  src = fetchFromGitHub {
    owner = "PixelSergey";
    repo = "meow";
    rev = "v${version}";
    hash = "sha256-PB871c137uxxPaYbV6NB8kECVUrQWTeVz0ciVLHrph4=";
  };
```

File: pkgs/by-name/me/meow/Cargo.toml.patch
Comment (Aleksanaa) [inline pkgs/by-name/me/meow/Cargo.toml.patch:5]: What's the reason of this?

File: pkgs/by-name/me/meow/package.nix
Comment (Aleksanaa) [inline pkgs/by-name/me/meow/package.nix:31]: missing `mainProgram`

File: pkgs/by-name/me/meow/package.nix
Comment (PixelSergey) [reply pkgs/by-name/me/meow/package.nix]: The package does not build without the `pname` parameter. Suggestion partially implemented in baff1d

File: pkgs/by-name/me/meow/Cargo.toml.patch
Comment (PixelSergey) [reply pkgs/by-name/me/meow/Cargo.toml.patch:5]: I want the command-line application to be run using the `meow` command, not using the `meow-cli` command.
I could not find a way to do this without patching the Cargo.toml file; changing it in the original file would not work since the Cargo package name `meow` is already taken and I would not be able to run `cargo publish`.

If you have a better solution to this, I would really appreciate it :)

File: pkgs/by-name/me/meow/package.nix
Comment (PixelSergey) [reply pkgs/by-name/me/meow/package.nix:29]: Thanks, fixed

File: pkgs/by-name/me/meow/package.nix
Comment (kilianar) [reply pkgs/by-name/me/meow/package.nix]: You're absolutely right—the `pname` attribute is still necessary. I mistakenly removed it. Apologies for the oversight.

File: pkgs/by-name/me/meow/Cargo.toml.patch
Comment (Aleksanaa) [reply pkgs/by-name/me/meow/Cargo.toml.patch:5]:
```
  postInstall = ''
    ln -s $out/bin/meow-cli $out/bin/meow
  '';
```

Also you are the upstream so you decide whatever upstream ships, but don't do dirty tricks here

File: nixos/doc/manual/release-notes/rl-2411.section.md
Comment (PixelSergey) [reply nixos/doc/manual/release-notes/rl-2411.section.md]: Fixed

File: pkgs/by-name/me/meow/Cargo.toml.patch
Comment (PixelSergey) [reply pkgs/by-name/me/meow/Cargo.toml.patch]: Omg thanks, I was pulling my hair out cause I couldn't get `postInstall` to work. This works now :D

**Review by Aleksanaa — APPROVED** [review]
Please squash into two commits:

maintainers: add pixelsergey
meow: init at 2.1.3

**Comment by PixelSergey** [comment]
> Please squash into two commits:
> 
> maintainers: add pixelsergey meow: init at 2.1.3

Squashed :+1:

**Review by Aleksanaa — APPROVED** [review]

**Comment by PixelSergey** [comment]
@Aleksanaa is there anything else that needs to be done here? Who is going to merge this PR?

**Comment by Aleksanaa** [comment]
I'll merge it when ofborg is done.

**Review by kilianar — APPROVED** [review]

**Review by yayayayaka — APPROVED** [review]
meow of approval

**Pull request merged**
